src/components/TagDropdown.tsx

```diff
@@ -0,0 +1,156 @@
+        }
+    };
+
+    // Controls tag selection
+    function handleSelectedTags(tag: string) {
+        // if tag is already selected, deselect it and remove from selectedTags
+        if (selectedTags.includes(tag)) {
+            const toRemove = selectedTags.indexOf(tag);
+            setSelectedTags(() => [
+                ...selectedTags.slice(0, toRemove),
+                ...selectedTags.slice(toRemove + 1),
+            ]);
+        } else {
+            setSelectedTags([...selectedTags, tag]);
+        }
+    }
+
+    function deselectTag(tag: string) {
+        const toRemove = selectedTags.indexOf(tag);
+        setSelectedTags(() => [
+            ...selectedTags.slice(0, toRemove),
+            ...selectedTags.slice(toRemove + 1),
+        ]);
+    }
+
+    function addTag(tag: string) {
+        // If tag is in props.tags, add tag to selectedTags array
+        // Else (new tag), add tag to props.tags (large tag array), probably requires request to db, then add tag to selectedTags
+        // ?? QUESTION: do we actually want a New tag to be added every time someone hits enter
+        if (!props.tags.includes(tag)) {
+            props.tags.push(tag);
+            // console.log("pushed tag");
+            // console.log(props.tags);
+        } 
+        setSelectedTags([...selectedTags, tag]);
+        // clear search bar
+        setSearchInput("");
+    }
+
+    return (
+        <div className="flex flex-col p-4 gap-[10px]">
+            <div className="border-2 relative">
+                {searchInput != "" && (
+                    <button 
+                        onClick={handleAdd}
+                        className="absolute right-0 py-3 px-4"
+                    >
+                        Add tag
+                    </button>)}
+                <input
+                    onClick={handleClick}
+                    onInput={handleSearch}
+                    onKeyUp={handleEnter}
+                    value={searchInput}
+                    placeholder="tag name"
+                    className="flex flex-col justify-center pl-2 border-2 rounded-lg h-[46px] w-[100%] text-black"
+                />
+            </div>
+            {display && (
+                <div className="border-2 h-[144px] rounded-xl overflow-auto">
+                    {(!searchInput ? props.tags : filteredTags).map((tag) => (
+                        <div
+                            key={tag}
+                            onClick={() => handleSelectedTags(tag)}
+                            className={`pl-2 py-2 hover:font-bold hover:bg-violet-600 active:bg-violet-600 ${selectedTags.includes(tag) && 'bg-violet-600 font-bold'}`}
+                        >
+                            <p>{tag}</p>
+                        </div>
+                    ))}
+                </div>
+            )}
+            {/* if display==true AND at least one tag is selected, show the div*/}
+            {display && selectedTags.length > 0 && (
```

Comment:
I suspect we will want to know which tags are applied to a given item even when the dropdown is closed. Could we remove `display` from this condition so that if there are selected tags they are always visible?